### ISSUE TITLE
constitutional-supremacy and continuous-compliance match a script by name and never read what it runs

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#104]
**Version:** 7.21.0
**Laws:** `constitutional-supremacy-doctrine`, `continuous-compliance-doctrine`

## What happens

```
No constitutional law enforcement scripts
No automated compliance checking scripts
```

Both are decided by looking up a key and never reading its value:

```js
// checkers/sacred-laws/constitutional-supremacy.js:60
if (!packageJson.scripts?.['check:laws'] &&
    !packageJson.scripts?.['audit:constitutional']) {
  violations.push('No constitutional law enforcement scripts');
}
```

`continuous-compliance-doctrine` accepts a longer list — `check:laws`,
`audit:constitutional`, `compliance:check`, `quality:gate`,
`report:compliance`, `report:quality` — and reads the values of none of them.

Our package.json:

```jsonc
"scripts": {
  "audit":  "ruleofcode audit --mode=full",
  "verify": "nx format:check && nx lint web && nx lint web-e2e && nx test web && nx run web:finalize && nx e2e web-e2e"
}
```

`audit` runs your CLI. `verify` runs it inside the full gate, and the pre-push
hook runs `verify` on every push, so the constitution is enforced on every commit
that leaves this repository. Both laws report that no such script exists.

## Why renaming would be the wrong fix

Adding `"check:laws": "npm run audit"` would clear both findings in ten seconds
and change nothing about what runs. That is the kind of edit this tool exists to
argue against: a number moves, a gate does not appear. We would rather leave the
laws red and report this.

There is also a smaller cost to the name-matching approach: it teaches a
convention that only this tool knows. An adopter who already has `audit`,
`lint`, `test` and `verify` gets told to add a seventh alias whose only reader is
the checker.

## Suggested direction

Read the command, not the key. The script that satisfies these laws is
identifiable by what it invokes:

```js
const runsRoc = cmd => /\b(ruleofcode|roc)\b[^&|]*\baudit\b/.test(cmd);
const satisfied = Object.values(packageJson.scripts ?? {}).some(runsRoc);
```

That accepts `audit`, `check:laws`, `precommit`, `verify`, or a script that
chains it among other steps — which is how it usually appears in a real
repository, since the audit is one stage of a gate rather than a command anyone
runs alone.

Keep the name list as a fallback for the case where the script shells out to
something the regex cannot see, and keep the suggestion string — but generate it
from whichever check failed, the way 7.20.0 made the branch-name advice come from
the accepted set.

## Related

Same shape as #96: a law asking again for something the project has already
answered elsewhere, in a place the checker does not look.
